django/db/models/fields/related.py

```diff
@@ -298,10 +281,10 @@ def contribute_to_class(self, cls, name, virtual_only=False):
             self.rel.related_name = related_name
         other = self.rel.to
         if isinstance(other, six.string_types) or other._meta.pk is None:
-            def resolve_related_class(field, model, cls):
+            def resolve_related_class(cls, model, field):
                 field.rel.to = model
                 field.do_related_class(model, cls)
-            add_lazy_relation(cls, self, other, resolve_related_class)
+            lazy_related_operation(resolve_related_class, cls, other, field=self)
```

Comment:
Could also define the callbacks on the relevant class (e.g. as ``RelatedField.resolve_related_class``) rather than as single-use local functions.